### ISSUE TITLE
Mention integer length limitations in `number_input`

### ISSUE THREAD

**Issue by tnwei** [streamlit/docs#106]
**Link to doc page in question (if any):** https://docs.streamlit.io/en/stable/api.html#streamlit.number_input

**Name of the Streamlit feature whose docs need improvement:** `st.number_input`

**What you think the docs should say:**

Hi folks! Found out when I was copy-pasting the integers generated by `torch.seed()` into `st.number_input`, that the last few digits went to zero after enter is pressed. There appears to be a limit for how long an integer input can be, which is something we don't deal with in Python, given its capability to natively handle big integers.

This appears to be floating point behaviour where the number ran out of significands. Digging around in the code base, seems that the Python side of the [widget source code](https://github.com/streamlit/streamlit/blob/develop/lib/streamlit/elements/number_input.py) uses integers natively. Thus, I'm guessing that the limit to integer length is inherent to how Javascript stores the number internally before passing it to the Streamlit application. But I digress. 

Anyway, I thought it would be worth mentioning `number_input`'s limitation in handling big integers. As a workaround, users can use `st.text_input` and post-process the input values instead. This docs improvement should be relevant to folks building machine learning apps, where large ints can be involved in setting RNG seeds.

I'll put in a PR if the maintainers are OK w/ the proposed improvement.

**Comment by MathCatsAnd** [comment]
For my own edification, I confirmed `st.number_input` and `st.slider` both throw an error if too large of a number is passed to one of the creation parameters as @snehankekre said:

`x = st.slider('Enter a number', value=2**54-1, step=1)`
yields 
`StreamlitAPIException: max_value (18014398509481983) must be <= (1 << 53) - 1`

`x = st.number_input('Enter a number', value=-2**54+1, step=1)`
yields
`StreamlitAPIException: value (-18014398509481983) must be >= -((1 << 53) - 1)`

Manually assigning a value through session state or on the front end that is out of bounds just compromises the value without warning unless the limits are explicitly set as in:
`x = st.number_input('Enter a number', min_value=-2**53+1, max_value=2**53-1, key='key')`

I searched the docs for '53' and only came up with issue numbers in the changelog, so I don't think this has been added yet. Would it make sense to add a note within the docstring of `st.number_input` and `st.slider` at the top, between the summary and the parameters:

```
    .. note::
        Integer values larger in magnitude than `+/- (1<<53) - 1` cannot be stored or returned correctly by the widget due to
        limitations when serializing values between the Python server and Javascript client. Numbers larger than this must be
        represented as floats and will consequently lose precision.
```

I tweaked @tnwei's suggestion slightly to avoid any implication that conversion happens automatically and try to be clear that incorrect integers will be stored/returned instead.

**Comment by MathCatsAnd** [comment]
I took a stab at it. 😸

**Comment by MathCatsAnd** [comment]
@snehankekre, I think this can be closed now. 🙂

**Comment by sfc-gh-dmatthews** [comment]
Issued resolved with streamlit/streamlit#6724. Closing.
